File: src/main/java/training/practice/P009BinaryTreeLevelOrderTraversalPractice.java
package training.practice;

import java.util.List;

public class P009BinaryTreeLevelOrderTraversalPractice {
    static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int val) {
            this.val = val;
        }

        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    // TODO: return level order traversal result.
    public static List<List<Integer>> solve(TreeNode root) {
        throw new UnsupportedOperationException("TODO: implement P009BinaryTreeLevelOrderTraversalPractice.solve");
    }

    public static void main(String[] args) {
        TreeNode root = new TreeNode(3,
                new TreeNode(9),
                new TreeNode(20, new TreeNode(15), new TreeNode(7)));
        System.out.println(solve(root)); // expected: [[3], [9, 20], [15, 7]]
    }
}

File: training/python/practice/p009_binary_tree_level_order_traversal_practice.py
"""P009 Binary Tree Level Order Traversal Practice
TODO: implement solve(root)
"""


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def solve(root):
    raise NotImplementedError("TODO: implement p009 level order traversal")


if __name__ == '__main__':
    root = TreeNode(3)
    root.left = TreeNode(9)
    root.right = TreeNode(20, TreeNode(15), TreeNode(7))
    print(solve(root))  # expected: [[3], [9, 20], [15, 7]]
